libraries/plugins/witness/include/graphene/witness/witness.hpp

```diff
@@ -74,7 +71,37 @@ class witness_plugin : public graphene::app::plugin {
    inline const fc::flat_map< chain::witness_id_type, fc::optional<chain::public_key_type> >& get_witness_key_cache()
    { return _witness_key_cache; }
 
+   bool block_production_task_canceled()
```

Comment:
Please move non-trivial implementations into the .cpp file.

Reply:
ok sure ! Thanks !